File: THCS/2023_2024/PhuYen_2324/Bai1/BAI1.cpp
#include <iostream>
#include <fstream>
#include <vector>

using namespace std;

// Hàm để tính độ bền của một số nguyên không âm
int calculate_persistence(int n) {
    if (n < 10) {
        return 0; // Độ bền là 0 nếu số có một chữ số
    }

    int product = 1;
    while (n > 0) {
        product *= (n % 10); // Tính tích các chữ số
        n /= 10;
    }

    return calculate_persistence(product) + 1; // Đệ quy cộng thêm 1
}

int main() {
    ifstream infile("BAI1.INP"); // Mở file để đọc
    ofstream outfile("BAI1.OUT"); // Mở file để ghi

    if (!infile) {
        cerr << "Không thể mở file BAI1.INP" << endl;
        return 1;
    }

    vector<int> results;
    int n;

    // Đọc từng số từ file
    while (infile >> n) {
        int persistence = calculate_persistence(n); // Tính độ bền
        results.push_back(persistence); // Lưu kết quả
    }

    // Ghi kết quả ra file
    for (int result : results) {
        outfile << result << endl;
    }

    infile.close(); // Đóng file đọc
    outfile.close(); // Đóng file ghi

    return 0;
}

File: THCS/2023_2024/PhuYen_2324/Bai1/BAI1.py
def calculate_persistence(n):
    # Nếu số có một chữ số, độ bền là 0
    if n < 10:
        return 0
    
    # Tính tích các chữ số của n
    product = 1
    while n > 0:
        product *= n % 10
        n //= 10
    
    # Đệ quy để tính độ bền
    return calculate_persistence(product) + 1

# Đọc dữ liệu từ file
with open('BAI1.INP', 'r') as infile:
    lines = infile.readlines()

# Khởi tạo danh sách để lưu kết quả
results = []

# Xử lý từng số trong file
for line in lines:
    n = int(line.strip())
    persistence = calculate_persistence(n)
    results.append(persistence)

# Ghi kết quả ra file
with open('BAI1.OUT', 'w') as outfile:
    for result in results:
        outfile.write(str(result) + '\n')
